x-pack/plugins/cases/server/client/cases/get.ts

```diff
@@ -97,14 +120,36 @@ export const get = async ({
 /**
  * Retrieves the tags from all the cases.
  */
-export async function getTags({
-  savedObjectsClient: soClient,
-  caseService,
-  logger,
-}: CasesClientArgs): Promise<string[]> {
+
+export async function getTags(
+  params: AllTagsFindRequest,
+  clientArgs: CasesClientArgs
+): Promise<string[]> {
+  const {
+    savedObjectsClient: soClient,
+    caseService,
+    logger,
+    authorization: auth,
+    auditLogger,
+  } = clientArgs;
+
   try {
+    const queryParams = pipe(
+      excess(AllTagsFindRequestRt).decode(params),
+      fold(throwErrors(Boom.badRequest), identity)
+    );
+
+    const { filter: authorizationFilter } = await getAuthorizationFilter({
+      authorization: auth,
+      operation: Operations.getTags,
+      auditLogger,
+    });
+
+    const filter = combineAuthorizedAndOwnerFilter(queryParams.owner, authorizationFilter);
```

Comment:
I think we need to call `ensureSavedObjectsAreAuthorized` for each of the cases that are returned right?

And then also the `logSuccessfulAuthorization` after as well.